source/robot_model/src/Model.cpp

```diff
@@ -1,21 +1,39 @@
-#include <iostream>
+#include <regex>
+#include <set>
 #include <pinocchio/algorithm/frames.hpp>
 #include <pinocchio/algorithm/joint-configuration.hpp>
 #include "robot_model/Model.hpp"
 #include "robot_model/exceptions/FrameNotFoundException.hpp"
 #include "robot_model/exceptions/InverseKinematicsNotConvergingException.hpp"
 #include "robot_model/exceptions/InvalidJointStateSizeException.hpp"
+#include "robot_model/exceptions/CollisionGeometryException.hpp" 
 
 namespace robot_model {
-Model::Model(const std::string& robot_name, const std::string& urdf_path) :
+Model::Model(const std::string& robot_name, 
+             const std::string& urdf_path,
+             const std::optional<std::function<std::string(const std::string&)>>& meshloader_callback
+             ):
     robot_name_(std::make_shared<state_representation::Parameter<std::string>>("robot_name", robot_name)),
-    urdf_path_(std::make_shared<state_representation::Parameter<std::string>>("urdf_path", urdf_path)) {
+    urdf_path_(std::make_shared<state_representation::Parameter<std::string>>("urdf_path", urdf_path)),
+    meshloader_callback_(meshloader_callback)
+    {
+  this->init_model(true);
+}
+
+Model::Model(const std::string& robot_name, 
+             const std::string& urdf_path
+             ):
+    robot_name_(std::make_shared<state_representation::Parameter<std::string>>("robot_name", robot_name)),
+    urdf_path_(std::make_shared<state_representation::Parameter<std::string>>("urdf_path", urdf_path))
+    {
   this->init_model();
 }
 
 Model::Model(const Model& model) :
     robot_name_(model.robot_name_),
-    urdf_path_(model.urdf_path_) {
+    urdf_path_(model.urdf_path_), 
+    meshloader_callback_(model.meshloader_callback_)
+    {
   this->init_model();
```

Comment:
This should now do `init_model(true)` no?

Reply:
Indeed, I'll update the docs for that too. Based on our current constructor design approach, we cannot defferentiate between which of the model constructor is being copied. With that, we can make the copy constructor applicable to the case where the load geometries is actually needed. 

Any other suggestions on this?

Reply:
Ouf that's true, my feedback was wrong again, I thought this was the constructor not the copy constructor. But this comes as a major drawback of the current design now that you are pointing it out. A copy constructor that doesnt copy everything is not good IMO. Can we infer if we need to load geometries in another way?

Reply:
One idea that I have in mind to provide a `load_geometries` flag and with that the following constructors are possible:

`Model("robot", "/path/to/urdf") # simple case`
`Model("robot", "/path/to/urdf", true) # second constructor`
`Model("robot", "/path/to/urdf", true, callback_function) # second constructor`

but then do we really need two constructors in that case? as both are the same in that scenario
`Model("robot", "/path/to/urdf") # simple case`
`Model("robot", "/path/to/urdf", false) # second constructor`

Moreover, we can then check in the copy constructor if `load_geometries` is true or false and accordingly we can initialize correctly the model. 

Another idea is to check if the callback is provided in the copy constructor and if it is, set the load_geometries to true and false otherwise, but then there is a third scenario where the file is well sanitized and we want to load the geometries without passing a callback then we will definitely need a flag to differentiate. 

What do you think?

Reply:
yes that would work! Just a small note, I don't think it is necessary in that case to provide an argument to the `init_model` as the attribute is set by the constructor and hence can be assessed by the `init_model`

Reply:
To be honest, this copy constructor is pretty inefficient since it just reconstructs everything. I have a hunch that the default copy constructor would have sufficed; all of the data members including pinocchio::Model / Data are STL or Eigen containers, i.e. no dynamically referenced shared memory. The only dynamically referenced memory are the shared pointer parameters `robot_name_` and `urdf_path_`, and it makes no sense for those to be shared pointer parameters because they are only ever used internally. 

If we replaced those two properties with plain strings, we could also drop the custom copy constructor entirely. If we did that, it would technically be a breaking ABI change because we modify inline functions and the data structure of the class.

Note that the original comments would also apply to the implementation of `swap` (which is also inefficient since it invokes `init_model` internally). I would personally drop `swap` and `operator=` definitions and let the compiler do the default business.

Reply:
I didn't see this comment until now, do you still want this to be addressed?

Reply:
It will be addressed in this issue over [here](https://github.com/aica-technology/control-libraries/issues/174) as it can be scoped separately to integrate collision detection.